### ISSUE TITLE
Full node sync stuck on infinite requesting a block

### ISSUE THREAD

**Issue by garyyu** [mimblewimble/grin#1684]
```
Oct 07 18:59:50.854 DEBG Received block 09508fc2 at 124823 from 185.80.129.70:13414, inputs: 0, outputs: 1, kernels: 1, going to process.
Oct 07 18:59:50.859 DEBG pipe: process_block 09508fc2 at 124823 with 0 inputs, 1 outputs, 1 kernels
Oct 07 18:59:50.901 DEBG Error returned, discarding txhashset extension: Store Error: NotFoundErr("Block sums for block: 17006ba7") 
 Cause: DB Not Found Error: Block sums for block: 17006ba7 
 Backtrace: 
Oct 07 18:59:50.902 INFO Rejected block 09508fc2 at 124823: Error { inner: NotFoundErr("Block sums for block: 17006ba7")

Store Error: NotFoundErr("Block sums for block: 17006ba7") }
Oct 07 18:59:50.902 DEBG pipe: process_block 13d7f6a3 at 124896 with 0 inputs, 1 outputs, 1 kernels
Oct 07 18:59:50.902 DEBG adapter: process_block: block 09508fc2 refused by chain: Other Error: Error { inner: NotFoundErr("Block sums for block: 17006ba7")

Store Error: NotFoundErr("Block sums for block: 17006ba7") }
...
Oct 07 18:59:51.724 DEBG Received block 09508fc2 at 124823 from 185.80.129.70:13414, inputs: 0, outputs: 1, kernels: 1, going to process.
Oct 07 18:59:51.724 DEBG pipe: process_block 09508fc2 at 124823 with 0 inputs, 1 outputs, 1 kernels
Oct 07 18:59:51.724 DEBG Block 09508fc2 at 124823 is unfit at this time: already known in cache
...
Oct 08 11:50:55.875 DEBG body_sync: body_head - 065e2a03, 124823, header_head - 0ef763f9, 128760, sync_head - 0ef763f9, 128760
Oct 08 11:50:55.972 DEBG block_sync: 124823/128760 requesting blocks [09508fc2] from 17 peers
Oct 08 11:50:55.972 DEBG Requesting block 09508fc2 from peer 139.162.168.18:13414.
Oct 08 11:50:56.126 DEBG handle_payload: received block: msg_len: 1260
Oct 08 11:50:56.126 DEBG Received block 09508fc2 at 124823 from 139.162.168.18:13414, inputs: 0, outputs: 1, kernels: 1, going to process.
Oct 08 11:50:56.126 DEBG pipe: process_block 09508fc2 at 124823 with 0 inputs, 1 outputs, 1 kernels
Oct 08 11:50:56.126 DEBG Block 09508fc2 at 124823 is unfit at this time: already known in cache
Oct 08 11:50:56.126 DEBG adapter: process_block: block 09508fc2 refused by chain: Block is unfit: already known in cache

```

A simple fix solution: We need handle `NotFoundErr` and not add this block into the `cache`, to avoid infinite `already known in cache`.

But I'm not sure if the new received block 09508fc2 will still have problem of `NotFoundErr`.  

@antiochp what's the meaning of this error? I remember I saw it before, several times.

**Comment by antiochp** [comment]
Is this an archival node that's been running for a long time?
We probably have "old" archival data in there, blocks that do not have their associated block_sums.

This probably slipped though in testing because we are not focusing on archival mode nodes.

This is also an old block (5,000 blocks prior to current head) - so this is an old archival node stuck syncing old data?

**Comment by ignopeverell** [comment]
I have a feeling this is fixed, right?

**Comment by garyyu** [comment]
Pls keep for a while, I want to give a bit "improve" on this, not a fix but better to avoid this infinite loop which consume the Grin network bandwidth for nothing.

**Comment by antiochp** [comment]
I believe this is fixed in #2009.
We got rid of the cache invalidation problems by getting rid of the cache entirely...
